workspace/read-manifest/src/catalogs.ts

```diff
@@ -0,0 +1,65 @@
+import { InvalidWorkspaceManifestError } from './errors/InvalidWorkspaceManifestError'
+
+export interface WorkspaceNamedCatalogs {
+  readonly [catalogName: string]: WorkspaceCatalog
+}
+
+export interface WorkspaceCatalog {
+  readonly [dependencyName: string]: string
+}
+
+export function assertValidWorkspaceManifestCatalog (manifest: { packages?: readonly string[], catalog?: unknown }): asserts manifest is { catalog?: WorkspaceCatalog } {
+  if (manifest.catalog == null) {
+    return
+  }
+
+  if (Array.isArray(manifest.catalog)) {
+    throw new InvalidWorkspaceManifestError('Expected catalog field to be an object, but found - array')
+  }
+
+  if (typeof manifest.catalog !== 'object') {
+    throw new InvalidWorkspaceManifestError(`Expected catalog field to be an object, but found - ${typeof manifest.catalog}`)
+  }
+
+  for (const [alias, specifier] of Object.entries(manifest.catalog)) {
+    if (typeof specifier !== 'string') {
```

Comment:
I briefly considered supporting integers. For example:

```yml
catalog:
  react: 18
```

While I'd personally use `^18.3.0`, `18` is technically a valid semver. Currently, users would have to write:

```yml
catalog:
  react: "18"
```

I think we should support plain integers as strings. Does that seem reasonable? I can do that in a following PR if so.

Reply:
no, let's only support it as string. Otherwise it becomes confusing as 18.10 isn't bigger than 18.2 according to semver.

Reply:
I definitely wouldn't support floats for the reason you mentioned. But yeah, even supporting just integers becomes a slippery slope. Okay, let's just do strings. 🙂